drivers/counter/counter_ll_stm32_rtc.c

```diff
@@ -181,12 +186,25 @@ static int rtc_stm32_stop(const struct device *dev)
 }
 
 
-static uint32_t rtc_stm32_read(const struct device *dev)
+#ifndef CONFIG_COUNTER_RTC_STM32_SUBSECONDS_BASED
+static uint32_t
+#else
+static uint64_t
+#endif
```

Comment:
I suggest you define a 32 or 64-bit type once at the top depending on the subsecond config, and then use this type each time you need it instead of having an #Ifdef/#else/#endif each time. That will reduce the number of different cases and ease future maintenance.

Reply:
Very good suggestion, thanks. Done.